portfolio/src/main/webapp/script.js

```diff
@@ -71,13 +71,17 @@ function contact() {
 	console.log('Contact!'); 
 }
 
+function updateNumber() {
+    getComments();
+}
+
 function getComments() {
-	fetch('/data').then(response => response.json()).then((comments) => {
-	 	const bodyElement = document.body;
-    console.log('Comments: ', comments);
-    bodyElement.innerHTML += 'Here are all the comments: <br>';
-    for (index in comments) {
-      bodyElement.innerHTML += comments[index] + '<br>';
-    }
+	fetch('/data?max-num=' + (document.getElementById("numComments").value)).then(response => response.json()).then((comments) => {
+	  const bodyElement = document.body;
+      console.log('Comments: ', comments);
+      bodyElement.innerHTML += 'Here are all the comments: <br>';
+      for (comment of comments) {
+        bodyElement.innerHTML += comment.propertyMap.text + '<br>';
```

Comment:
This is an interesting change (that I'm fine with in this PR). It's generally a good practice to separate your wire formats and your storage formats:
1) You're now pushing more data than the client needs over the wire (in this case it's only timestamp, but what if it expands?).
2) What if you decide to port your storage layer from DatastoreService to BigTable? Now all of your client code will need to change, as well.

Reply:
ah this is a great point :) I changed the underlying data servlet to make it more generalized!